src/Tools/dotnet-dump/ExtensionCommands/TaskStateCommand.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Diagnostics.Repl;
+
+namespace Microsoft.Diagnostic.Tools.Dump.ExtensionCommands
+{
+    [Command(Name = "taskstate", AliasExpansion = "TaskState", Help = "Display a Task state in a human readable format.")]
+    [CommandAlias(Name = "tks")]
+    public class TaskStateCommand : ExtensionCommandBase
+    {
+        [Argument(Help = "The Task instance address.")]
+        public string Address { get; set; }
+
+        [Option(Name = "--value", Help = "<value> is the value of a Task m_stateFlags field.")]
+        [OptionAlias(Name = "-v")]
+        public ulong? Value { get; set; }
+
+
+        public override void Invoke()
+        {
+            if (string.IsNullOrEmpty(Address) && !Value.HasValue)
+            {
+                WriteLine("Missing Task reference address or state value..." + Environment.NewLine);
+                return;
+            }
+
+            ulong stateFlag = Value.GetValueOrDefault();
+
+            // access the Task state field if the flag is not given as a parameter
+            if (!Value.HasValue)
+            {
+                // skip 0x or leading 0000 if needed 
+                if (Address.StartsWith("0x"))
```

Comment:
You don't have to do the string address parsing your self. You can make "Address" a ulong.  Right now it will convert a just hex number, but ulong allow future expression parsing.

Reply:
Unfortunately, the parsing is not very smart because it expects that hex numbers are prefixed by 0x
![image](https://user-images.githubusercontent.com/10163781/90738993-d2714380-e2ce-11ea-8724-ff45d008a9bb.png)
But all addresses displayed by sos commands don't have this 0x prefix. 
So in term of usability, it is simpler to support direct copy/paste from other commands result than type 0x then copy/paste

Reply:
It would be better for me to try to fix the parsing problem so all commands benefit.  Having every new command do this kind of parsing isn't very efficient or clean. I created issue https://github.com/dotnet/diagnostics/issues/1460 to track this.

Reply:
Good idea!
I will update the command when it will be fixed: would you merge the current implementation?